File: QuSoC/QuSoC/apps/Counter/source/Counter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SOCCounter
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            uint counter = 0;
            while (true)
            {
                counter++;
                SOC.Instance.Counter = counter;
            }
        }
    }
}

File: QuSoC/QuSoC/apps/Counter/firmware/Counter.cpp
#include "Counter.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace Counter
{
	void Firmware::EntryPoint()
	{
		uint32_t counter = 0;
		while(true)
		{
			(counter++);
			Counter_SOC_Counter = counter;
		}
	}
}
